packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -1113,11 +1113,22 @@ export const openPortfolio = (async (
       await registerNobleForwardingAccount(sender, dest, forwarding, traceP);
     }
 
-    try {
-      await rebalance(orch, ctxI, seat, offerArgs, kit);
-    } catch (err) {
-      traceP('⚠️ rebalance failed', err);
-      if (!seat.hasExited()) seat.fail(err);
+    const proposal = seat.getProposal() as ProposalType['rebalance'];
+    const deposit = (proposal.give as { Deposit?: NatAmount }).Deposit;
+    if (deposit) {
```

Comment:
_another_ cast? I don't understand why that would be needed.

```suggestion
    if ('Deposit' in proposal.give) {
      const { Deposit: deposit } = proposal.give;
```

Reply:
Bad typing on my part; fixed.